spec/src/main/asciidoc/metrics.asciidoc

```diff
@@ -0,0 +1,235 @@
+//
+// Copyright (c) 2018 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Andrew Rouse
+
+== Integration with Microprofile Metrics
+
+When Microprofile Fault Tolerance and Microprofile Metrics are used together, metrics are automatically added for each of
+the methods annotated with a `@Retry`, `@Timeout`, `@CircuitBreaker` or `@Bulkhead` annotation.
+
+=== Names
+
+The automatically added metrics follow a consistent pattern which includes the fully qualified name of the annotated method.
+In the tables below, the placeholder `<name>` should be replaced by the fully qualified method name with all `.` characters replaced with `_`.
+
+=== Metrics added for `@Retry`, `@Timeout`, `@CircuitBreaker` and `@Bulkhead`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_invocations_total`
+| Counter | None
+| The number of times the method was called
+
+|`ft_<name>_invocations_failed`
+| Counter | None
+| The number of times the method was called and, after all Fault Tolerance actions had been processed, threw a `Throwable`
+|===
+
+=== Metrics added for `@Retry`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_retry_calls_succeeded_not_retried_total`
+| Counter | None
+| The number of times the method was called and succeeded without retrying
+
+|`ft_<name>_retry_calls_succeeded_retried_total`
+| Counter | None
+| The number of times the method was called and succeeded after retrying at least once
+
+|`ft_<name>_retry_calls_failed_total`
+| Counter | None
+| The number of times the method was called and ultimately failed after retrying
+
+|`ft_<name>_retry_retries_total`
+| Counter | None
+| The total number of times the method was retried
+|===
+
+Note that the sum of `ft_<name>_retry_calls_succeeded_not_retried_total`, `ft_<name>_retry_calls_succeeded_retried_total` and `ft_<name>_retry_calls_failed_total` will give the total number of calls for which the Retry logic was run.
+
+=== Metrics added for `@Timeout`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_timeout_execution_duration`
+| Histogram | Nanoseconds
+| Histogram of execution times for the method
```

Comment:
note that, in the prometheus format, when a metric is in nanoseconds it will automatically be converted to seconds.